src/hooks/useGrantContract.ts

```diff
@@ -0,0 +1,46 @@
+import { ethers } from 'ethers';
+import { TokenVestingController__factory } from 'contracts/types';
+
+export function useGrantContract(address: string) {
```

Comment:
Please refer to other hooks on how to properly connect to a smart contract, such as by using `useWeb3React()` to get the provider.